doc/sourmash-internals.md

```diff
@@ -0,0 +1,648 @@
+* an optional `name`
+* an optional `filename`
+* a license (currently must be CC0);
+* an `md5sum(...)` method that returns a hash of the sketch.
+
+For now, we tend to refer to signatures and sketches interchangeably,
+because they are almost entirely 1:1 in the code base (but see [sourmash#616](https://github.com/sourmash-bio/sourmash/issues/616)).
+
+The default signature interchange/serialization format is JSON, optionally
+gzipped. This format is written and read by Rust code.
+
+In general, a lot of effort in sourmash is spent managing collections of
+signatures _before_ actually doing comparisons with them; see manifests,
+and `Index` objects, below.
+
+### Making sketches
+
+Sketches are produced by hashing k-mers with murmurhash and then
+keeping either the lowest `num` hashes (for MinHashes sketches) or
+keeping all hashes below `2**64 / scaled` (for FracMinHash sketches).
+This has the effect of selecting approximately one hash for every
+`scaled` k-mers - so, when sketching a set of 100,000 distinct k-mers,
+a scaled value of 1,000 would yield approximately 100 hashes to be
+retained in the sketch.
+
+The default MinHash sketches use parameters so that they are
+compatible with mash sketches.
+
+See [utils/compute-dna-mh-another-way.py](https://github.com/sourmash-bio/sourmash/blob/latest/utils/compute-dna-mh-another-way.py) for details on how k-mers are
+hashed.
+
+Note that if hashes are produced some other way (with a different hash
+function) or from some source other than DNA, sourmash can still work
+with them; only `sourmash sketch` does input checking.
+
+### Compatibility checking
+
+The point of the signatures and sketches is to enable certain kinds of
+rapid comparisons - Jaccard similarity and overlap analysis,
+specifically. However, these comparisons can only be done between
+compatible sketches.
+
+Here, "compatible" means -
+* the same MurmurHash seed (default 42);
+* the same k-mer size/ksize (see k-mer sizes, below);
+* the same molecule type (see molecule types, below);
+* the same `num` or `scaled` (although see [this downsampling discussion](api-example.md#downsampling-signatures), and the next two sections);
+
+sourmash uses selectors (`Index.select(...)`) to select sketches with
+compatible ksizes, molecule types, and sketch types.
+
+### Scaled (FracMinHash) sketches support similarity and containment
+
+Per our discussion in [Irber et al., 2022](https://www.biorxiv.org/content/10.1101/2022.01.11.475838v2), FracMinHash sketches can always be compared
+by downsampling to the max of the two scaled values.  (This is not always
+true of sketches stored in indexed collections, e.g. SBTs; see [sourmash#1799](https://github.com/sourmash-bio/sourmash/issues/1799).)
+
+In practice, sourmash does all necessary downsampling dynamically, but
+returns the original sketches. This means that (for example) you can
+do a low-resolution/high-scaled large scale search by specifying a
```

Comment:
Could "large scale" be changed to some synonym? Big, comprehensive, massive, sizable, huge?

```suggestion
do a low-resolution/high-scaled extensive search by specifying a
```

Reply:
updated in fbbcee56